lib/shoulda/matchers/active_record/association_matcher.rb

```diff
@@ -774,6 +774,29 @@ def have_one(name)
       #         class_name('City')
       #     end
       #
+      # ##### join_table
+      #
+      # Use `join_table` to test usage of the `:join_table` option. This
+      # asserts that the table you're referring to actually exists.
+      #
+      #     class Person < ActiveRecord::Base
+      #       has_and_belongs_to_many :issues, join_table: 'people_tickets'
+      #     end
+      #
+      #     # RSpec
+      #     describe Person do
+      #       it do
+      #         should have_and_belong_to_many(:issues).
+      #           join_table('people_tickets')
```

Comment:
In [this line](https://github.com/thoughtbot/shoulda-matchers/pull/704/files#diff-0a438a581d903b7f45fdb3811e6e9291R783) I defined a custom `join_table`.

My rationale is that this option is more used when you have a custom join table, I don't think we need to test join table on a default HABTM.